src/plugins/intel_gpu/src/graph/eltwise.cpp

```diff
@@ -376,10 +376,16 @@ eltwise_inst::typed_primitive_inst(network& network, eltwise_node const& node) :
                 }
             }
 
-            for (size_t d = 0; d < input0_pshape.size(); ++d) {
-                bool sizes_equal = input0_pshape[d] == input_pshape[d];
+            auto base_pshape = input0_pshape;
+            if (prim->broadcast_spec == ov::op::AutoBroadcastType::NUMPY &&
+                base_pshape.size() < input_pshape.size()) {
+                base_pshape.insert(base_pshape.begin(), input_pshape.size() - base_pshape.size(), 1);
+            }
+
+            for (size_t d = 0; d < base_pshape.size(); ++d) {
+                bool sizes_equal = base_pshape[d] == input_pshape[d];
                 bool broadcast =
-                    (input0_pshape[d] == 1 || input_pshape[d] == 1) && (input0_pshape[d] != 1 || input_pshape[d] != 1);
+                    (base_pshape[d] == 1 || input_pshape[d] == 1) && (base_pshape[d] != 1 || input_pshape[d] != 1);
```

Comment:
could you check whether we have func test for such broadcast case? Is it disabled for GPU? If it is disabled, please enable it now. If it is not implemented, could you create one?

Reply:
This code can be reached at allow_new_shape_infer = true and static. In static eltwise func test with allow_new_shape_infer=false, 3d partial shape input became 4d tensor automatically. And creating static eltwise func test with allow_new_shape_infer = true needs to include specific primitive to enable allow_new_shape_infer like 'FullyConnectedCompressed'. (refer ProgramBuilder::requires_new_shape_infer())
Instead of adding such a complicated exceptional func test, I added extra unittest which 3d{2, 4, 2} input test and right-alignment-extended 4d{1, 2, 4, 2} input test has same result by NUMPY broadcasting. This would be more proper for original purpose which is right-alignment for NUMPY broadcast

Reply:
oh I see. could you add such description in commit message or comment of test code?

Reply:
Added some comments to give info at the test code.